### PR TITLE
import bom 

### PR DESCRIPTION
**Pull request by ring630**
This method reflects the real application better

### CODE REVIEW

File: pyaedt/edb_core/components.py [hunk 1]
```diff
@@ -1541,13 +1541,13 @@ def import_bom(
         delimiter : str, optional
             Value to use for the delimiter. The default is ``","``.
         refdes_col : int, optional
-            Column index of refdes.
+            Column index of refdes. The default is ``"0"``.
```
Comment (PipKat) [inline pyaedt/edb_core/components.py:1544]:
```suggestion
            Column index of reference designator. The default is ``"0"``.
```

File: pyaedt/edb_core/components.py [hunk 2]
```diff
@@ -1541,13 +1541,13 @@ def import_bom(
...
         part_name_col : int, optional
-             Column index of part name.
+             Column index of part name. The default is ``"1"``. Set to None if not exist.
```
Comment (PipKat) [inline pyaedt/edb_core/components.py:1546]:
```suggestion
             Column index of part name. The default is ``"1"``. The value is set to ``None`` if the
             column does not exist.
```

File: pyaedt/edb_core/components.py [hunk 3]
```diff
@@ -1541,13 +1541,13 @@ def import_bom(
...
         comp_type_col : int, optional
-            Column index of component type.
+            Column index of component type. The default is ``"2"``.
         value_col : int, optional
-            Column index of value.
+            Column index of value. The default is ``"3"``. Set to None if not exist.
```
Comment (PipKat) [inline pyaedt/edb_core/components.py:1550]:
```suggestion
            Column index of value. The default is ``"3"``. The value is set
            to ``None`` if the column does not exist.
```

File: pyaedt/edb_core/EDB_Data.py [hunk 4]
```diff
@@ -3121,6 +3121,15 @@ def partname(self):
         """
         return self.edbcomponent.GetComponentDef().GetName()
 
+    @partname.setter
+    def partname(self, name):
+        """Set component Part Name.
+
+        Returns
+        -------
+        """
```
Comment (MaxJPRey) [inline pyaedt/edb_core/EDB_Data.py:3130]:
```suggestion
        """Set component Part Name."""
```

File: pyaedt/edb_core/EDB_Data.py [hunk 5]
```diff
@@ -3121,6 +3121,15 @@ def partname(self):
...
+        return self.edbcomponent.GetComponentDef().SetName(name)
```
Comment (MaxJPRey) [inline pyaedt/edb_core/EDB_Data.py:3131]: @ring630 I am just double checking. Do you expect to return a value in the setter?
```suggestion
        self.edbcomponent.GetComponentDef().SetName(name)
```

File: pyaedt/edb_core/components.py [hunk 6]
```diff
@@ -1522,6 +1522,72 @@ def update_rlc_from_bom(
...
+        Parameters
+        ----------
+        bom_file : str
+            Full path to the BOM file, which is a delimited text file.
+        delimiter : str, optional
+            Value to use for the delimiter. The default is ``","``.
+        refdes_col : int, optional
+            Column index of refdes. The default is ``"0"``.
+        part_name_col : int, optional
+             Column index of part name. The default is ``"1"``. Set to None if not exist.
```
Comment (MaxJPRey) [inline pyaedt/edb_core/components.py:1546]:
```suggestion
             Column index of part name. The default is ``"1"``. Set to ``None`` if not exist.
```

File: pyaedt/edb_core/components.py [hunk 7]
```diff
@@ -1522,6 +1522,72 @@ def update_rlc_from_bom(
...
+            Column index of value. The default is ``"3"``. Set to None if not exist.
+        Returns
+        -------
+        bool
+        """
+        with open(bom_file, "r") as f:
+            lines = f.readlines()
+            unmount_comp_list = list(self.components.keys())
+            for l in lines[1:]:
+                print(l)
```
Comment (MaxJPRey) [inline pyaedt/edb_core/components.py:1559]: I would suggest to remove the `print`. If necessary, can we use the `logger` instead?
```suggestion
```

**Review by MaxJPRey — APPROVED** [review]
LGTM.

**Pull request merged**